pkg/controller/framework/controller.go

```diff
@@ -119,13 +154,8 @@ func (c *Controller) Requeue(obj interface{}) error {
 func (c *Controller) processLoop() {
 	for {
 		obj := c.config.Queue.Pop()
-		err := c.config.Process(obj)
-		if err != nil {
-			if c.config.RetryOnError {
-				// This is the safe way to re-enqueue.
-				c.config.Queue.AddIfNotPresent(obj)
-			}
-		}
+		// TODO: Dispatch it correctly.
+		c.workQueue[0] <- obj
```

Comment:
Yeah - I agree that in general it may not be correct, but I think a bunch of controllers can tolerate this.
And note that setting numbers of workers to 1 makes the order to be the original order again. So you will be able to configure it to preserve the original behavior (we may even make it a default behavior).
WDYT?